Reflect.ts

```diff
@@ -675,7 +675,7 @@ namespace Reflect {
 
         // Load global or shim versions of Map, Set, and WeakMap
         const functionPrototype = Object.getPrototypeOf(Function);
-        const usePolyfill = typeof process === "object" && process.env && process.env["REFLECT_METADATA_USE_MAP_POLYFILL"] === "true";
+        const usePolyfill = typeof process === "object" && process["env" + ""] && process["env" + ""]["REFLECT_METADATA_USE_MAP_POLYFILL"] === "true";

```

Comment:
when would `process` exist but `process.env` not?

wouldn't this be simpler?:
```suggestion
        const usePolyfill = typeof process === "object" && process.env.REFLECT_METADATA_USE_MAP_POLYFILL === "true";

```

Reply:
> when would `process` exist but `process.env` not?

Because who knows what globals might exist in a project that loads this? I'd rather not just break consumers for no reason.



> wouldn't this be simpler?:

No, `process.env.REFLECT_METADATA_USE_MAP_POLYFILL` still results in full env injection in webpack.

Reply:
wow, that's terrible. is there an issue filed on webpack? working around webpack issues in individual libraries doesn't scale :-/

Reply:
I haven't filed one, no. You're welcome to, if you're interested in following up. I'm happy to fix this here in the meantime.